server/src/main/java/com/flipkart/varadhi/db/VaradhiMetaStore.java

```diff
@@ -1,66 +1,251 @@
 
     public VaradhiMetaStore(CuratorFramework zkCurator) {
         this.zkMetaStore = new ZKMetaStore(zkCurator);
+        ensureEntityTypePathExists();
+    }
+
+    private void ensureEntityTypePathExists() {
+        this.ensureEntityTypePathExists(ORG);
+        this.ensureEntityTypePathExists(TEAM);
+        this.ensureEntityTypePathExists(PROJECT);
+        this.ensureEntityTypePathExists(TOPIC_RESOURCE);
+        this.ensureEntityTypePathExists(VARADHI_TOPIC);
+    }
+
+    public void ensureEntityTypePathExists(ZNodeKind zNodeKind) {
+        ZNode znode = ZNode.OfEntityType(zNodeKind);
+        if (!this.zkMetaStore.zkPathExist(znode)) {
+            this.zkMetaStore.createZNode(znode);
+        }
     }
 
     @Override
-    public TopicResource createTopicResource(TopicResource resource) {
-        String resourcePath = ZKPathUtils.getTopicResourcePath(resource.getProject(), resource.getName());
-        this.zkMetaStore.create(resource, resource.getVersion(), resourcePath);
-        return resource;
+    public void createOrg(Org org) {
+        ZNode znode = ZNode.OfOrg(org.getName());
+        this.zkMetaStore.createZNodeWithData(znode, org);
     }
 
     @Override
-    public boolean checkTopicResourceExists(String projectName, String topicName) {
-        String resourcePath = ZKPathUtils.getTopicResourcePath(projectName, topicName);
-        return this.zkMetaStore.exists(resourcePath);
+    public Org getOrg(String orgName) {
+        ZNode znode = ZNode.OfOrg(orgName);
+        return this.zkMetaStore.getZNodeDataAsPojo(znode, Org.class);
     }
 
     @Override
-    public TopicResource getTopicResource(String projectName, String resourceName) {
-        String resourcePath = ZKPathUtils.getTopicResourcePath(projectName, resourceName);
-        return this.zkMetaStore.get(resourcePath, TopicResource.class);
+    public boolean checkOrgExists(String orgName) {
+        ZNode znode = ZNode.OfOrg(orgName);
+        return this.zkMetaStore.zkPathExist(znode);
     }
 
     @Override
-    public VaradhiTopic createVaradhiTopic(VaradhiTopic varadhiTopic) {
-        String resourcePath = ZKPathUtils.getVaradhiTopicPath(varadhiTopic.getName());
-        this.zkMetaStore.create(varadhiTopic, varadhiTopic.getVersion(), resourcePath);
-        return varadhiTopic;
+    public void deleteOrg(String orgName) {
+        ZNode znode = ZNode.OfOrg(orgName);
+        this.zkMetaStore.deleteZNode(znode);
     }
 
     @Override
-    public boolean checkVaradhiTopicExists(String varadhiTopicName) {
-        String resourcePath = ZKPathUtils.getVaradhiTopicPath(varadhiTopicName);
-        return this.zkMetaStore.exists(resourcePath);
+    public List<Org> getOrgs() {
+        ZNode znode = ZNode.OfEntityType(ORG);
+        return this.zkMetaStore.listChildren(znode)
+                .stream()
+                .map(name -> {
+                    //TODO:: Discuss for Removing this handling. Discuss this exception handling may not be required if
```

Comment:
yes, fair assumption. we can fail the whole operation even if one element fails.